test/gtest/ucs/test_memtype_cache.cc

```diff
@@ -278,3 +291,69 @@ UCS_TEST_P(test_memtype_cache, diff_mem_types_diff_bufs_keep_mem) {
 
 INSTANTIATE_TEST_CASE_P(mem_type, test_memtype_cache,
                         ::testing::ValuesIn(mem_buffer::supported_mem_types()));
+
+class test_memtype_cache_deferred_create : public test_memtype_cache {
+protected:
+    virtual void init() {
+        /* do nothing */
+    }
+
+    void test_unknown_region_found(const mem_buffer &b) const {
+        if (b.mem_type() == UCS_MEMORY_TYPE_HOST) {
+            return;
+        }
+
+        test_ptr_found(b.ptr(), b.size(), UCS_MEMORY_TYPE_LAST);
+    }
+
+    void test_alloc_before_init(size_t buf_size, bool test_adjacent = false,
```

Comment:
ok, that looks better.
no need to have default values for params, pass them explicitly.

Reply:
done